### PR TITLE
fix: support uri/host/remote_addr

### LINKED ISSUE #2013: support uri & uris, host & hosts, remote_addr & remote_addrs
When we create a Route, the API supports `host/hosts` and `uri/uris` and `remote_addr/remote_addrs`. No matter API contains uri or uris, the Web should support all of them, we could use List component to show uris.

image

Currently, the Web supports `uris/hosts/remote_addrs`, and please have a check if Web could show `uri/host/remote_addr`.

### PR DESCRIPTION
**Pull request by Baoyuantop**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

When creating a route, if `Host, Path, and Remote Addrs` have a single value, send `uri/host/remote_addr` to the API, and send `uris/hosts/remote_addrs` if there are multiple values.

**Related issues**

resolve #2013 

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [ ] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Comment by liuxiran** [comment]
Hi @Baoyuantop please fix ci problem when you have time, thanks

**Comment by juzhiyuan** [comment]
@liuxiran Could you please give some test cases so we could test them in this PR?

**Comment by liuxiran** [comment]
test cases should cover @Baoyuantop 
1. create a route in UI with `uri` `remote_addr` `host`, Intercept response body to see if they are as expected
2. create a route in UI with `uris` `remote_addrs` `hosts`, Intercept response body to see if they are as expected
3. both of them need to test the configure data are as expected

cc @juzhiyuan

**Comment by juzhiyuan** [comment]
And if the origin uri is `test.com`, then you change it to `test.com, test1.com`, then it should pass `uris`.

**Comment by Baoyuantop** [comment]
> test cases should cover @Baoyuantop
> 
> 1. create a route in UI with `uri` `remote_addr` `host`, Intercept response body to see if they are as expected
> 2. create a route in UI with `uris` `remote_addrs` `hosts`, Intercept response body to see if they are as expected
> 3. both of them need to test the configure UI are as expected
> 
> cc @juzhiyuan

Can `Intercept response body` be replaced by `Intercept Raw Data` when created? @liuxiran

**Comment by juzhiyuan** [comment]
I think it's ok :)

**Comment by juzhiyuan** [comment]
cc @iamayushdas @guoqqqi to have a review when you have time

File: web/cypress/integration/route/create-route-both-use-uri-uris.spec.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    cy.contains('Route').click();
+    cy.get(selector.empty).should('be.visible');
+    cy.contains('Create').click();
+    cy.contains('Next').click().click();
+    cy.get(selector.name).type(data.name);
+    cy.get(selector.description).type(data.description);
+
+    cy.get(selector.hosts_0).type('0.0.0.0');
+    cy.get(selector.uris_0).type('/test');
+    cy.get(selector.remote_addrs_0).type('192.168.1.0');
```
Comment (guoqqqi) [inline web/cypress/integration/route/create-route-both-use-uri-uris.spec.js:69]: `.type(XXX)` These test data can be placed in `data`

File: web/cypress/integration/route/create-route-both-use-uri-uris.spec.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    cy.contains('Search').click();
+    cy.contains(data.name).siblings().contains('Configure').click();
+
+    cy.get('#status').should('have.class', 'ant-switch-checked');
+    cy.get(selector.addHost).click();
+    cy.get(selector.hosts_1).type('1.1.1.1');
+    cy.get(selector.addUri).click();
+    cy.get(selector.uris_1).type('/test1');
+    cy.get(selector.addRemoteAddr).click();
+    cy.get(selector.remote_addrs_1).type('192.168.1.1');
```
Comment (guoqqqi) [inline web/cypress/integration/route/create-route-both-use-uri-uris.spec.js:117]: Move to public data

File: web/cypress/integration/route/create-route-both-use-uri-uris.spec.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    cy.contains('Route').click();
+
+    cy.get(selector.nameSelector).type(data.name);
+    cy.contains('Search').click();
+    cy.contains(data.name).siblings().contains('More').click();
+    cy.contains('View').click();
+    cy.get(selector.drawer).should('be.visible');
+
+    cy.get(selector.monacoScroll).within(() => {
+      cy.contains('uri').should('exist');
```
Comment (LiteSun) [inline web/cypress/integration/route/create-route-both-use-uri-uris.spec.js:103]: It would be better to  verify the corresponding values are correct

Comment (Baoyuantop) [reply web/cypress/integration/route/create-route-both-use-uri-uris.spec.js:103]: I didn't find a way to get the value through the field in the `Row Data Editor`, any good suggestions?

In addition, is it possible to verify that the values are equal on the configuration page?

Comment (Baoyuantop) [reply web/cypress/integration/route/create-route-both-use-uri-uris.spec.js:103]: Done ~

**Comment by liuxiran** [comment]
@Baoyuantop please fix the conflict files first

File: web/cypress/integration/route/create-route-both-use-uri-uris.spec.js [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    cy.contains('View').click();
+    cy.get(selector.drawer).should('be.visible');
+
+    cy.get(selector.monacoScroll).within(() => {
+      cy.contains('uris').should('exist');
+      cy.contains('hosts').should('exist');
+      cy.contains('remote_addrs').should('exist');
+    });
+  });
+
```
Comment (liuxiran) [inline web/cypress/integration/route/create-route-both-use-uri-uris.spec.js:160]: please add cases to confirm the configure view render normally with `uri/host/remote_addr`
and `uris/hosts/remote_addrs`
@Baoyuantop

Comment (Baoyuantop) [reply web/cypress/integration/route/create-route-both-use-uri-uris.spec.js:161]: resolve @liuxiran

**Review by liuxiran — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

**Review by LiteSun — APPROVED** [review]

**Review by guoqqqi — APPROVED** [review]

**Review by iamayushdas — APPROVED** [review]

**Pull request merged**